### PR TITLE
Add Feature - Budget (Complete)

### LINKED ISSUE #135: Add Budget feature

### PR DESCRIPTION
**Pull request by jared98lyj**
1. Added Budget class for budget tracking
2. Added add_budget and edit_budget command and parsers
3. Added storage capabilites for budget (I/O when startup)
4. Integrated storage with StorageManager
5. Integrated changed to the respective java test classes

//TODO add view_budget and delete_budget commands

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/budgetparser/EditBudgetCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * Primary method that throws an {@code EditBudgetCommand} to be executed.
+     * @throws ParseException
+     */
+    public EditBudgetCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_BUDGET);
+
+        if (!ArgumentTokenizer.arePrefixesPresent(argMultimap, PREFIX_BUDGET)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddBudgetCommand.MESSAGE_USAGE));
```
Comment (glatiuden) [inline src/main/java/seedu/address/logic/parser/budgetparser/EditBudgetCommandParser.java]: This should be `EditBudgetCommand` instead?

**Review by glatiuden — CHANGES_REQUESTED** [review]
The rest of the codes LGTM, just slight changes to the incorrect `MESSAGE_USAGE` being printed.

File: src/main/java/seedu/address/logic/parser/budgetparser/EditBudgetCommandParser.java (hunk 1 above)
Comment (jared98lyj) [reply src/main/java/seedu/address/logic/parser/budgetparser/EditBudgetCommandParser.java]: Got it

**Review by glatiuden — DISMISSED** [review]
LGTM

**Comment by jared98lyj** [comment]
1. Added delete_budget utility
2. Fixed formatting errors
3. Updated User guide for budget commands

**Review by glatiuden — DISMISSED** [review]
LGTM

**Comment by jared98lyj** [comment]
1. Added view_budget command
2. Updated user guide

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 2]
```diff
@@ -116,6 +120,26 @@ public void saveAppointmentBook(ReadOnlyAppointmentBook appointmentBook, Path fi
...
+    }
+
+    @Override
+    public void saveBudgetBook(BudgetBook budgetBook) throws IOException {
+        if (budgetBook.hasBudget()) {
+            System.out.println("There");
+            budgetBookStorage.saveBudget(budgetBook.getBudget().getValue(),
+                    budgetBook.getBudget().getTotalCost());
+        } else {
+            System.out.println("Here");
```
Comment (glatiuden) [inline src/main/java/seedu/address/storage/StorageManager.java]: I think you forgotten to remove the debugging print statement

File: src/main/java/seedu/address/storage/StorageManager.java [hunk 3]
```diff
@@ -116,6 +120,26 @@ public void saveAppointmentBook(ReadOnlyAppointmentBook appointmentBook, Path fi
...
+    // ================ BudgetBook methods ==========================
+    @Override
+    public BudgetBook readBudgetBook() {
+        return this.budgetBookStorage.loadBudgetBook();
+    }
+
+    @Override
+    public void saveBudgetBook(BudgetBook budgetBook) throws IOException {
+        if (budgetBook.hasBudget()) {
+            System.out.println("There");
```
Comment (glatiuden) [inline src/main/java/seedu/address/storage/StorageManager.java]: I think you forgotten to remove the debugging print statement here as well

**Review by glatiuden — DISMISSED** [review]
The rest of the codes looks LGTM - if there's any new additional codes other than minor bug fixes, try to open a new PR instead as it's growing a bit too big.

File: src/main/java/seedu/address/storage/StorageManager.java (hunk 3 above)
Comment (jared98lyj) [reply src/main/java/seedu/address/storage/StorageManager.java]: Wah tks you are sharp :)))

**Review by glatiuden — APPROVED** [review]
LGTM

**Review by kingsleykuan — APPROVED** [review]
LGTM

**Pull request merged**